crates/core/src/yarn_fn/function_wrapping.rs

```diff
@@ -131,6 +143,13 @@ impl PartialEq for Box<dyn UntypedYarnFn + Send + Sync> {
 
 impl Eq for Box<dyn UntypedYarnFn + Send + Sync> {}
 
+#[macro_export]
+macro_rules! yarn_fn {
+    (($($param:ty),+) -> $ret:ty) => {
+        impl YarnFn<fn($($param),+) -> $ret, Out = $ret>
+    };
+}
+
```

Comment:
Re: your visibility problem.
Remove the `#[macro_export]` and add a `pub use yarn_fn;` right here **under** the macro declaration, then you can export it in the prelude.
I know, it's weird 😅 A byproduct of macros not being "hoisted" and not accepting visibility modifiers.

Reply:
`pub use <macro_name>` requires that `#[macro_rules]` is also specified :/